File: spring-boot-iot-alarm/src/main/java/com/ghlzm/iot/alarm/entity/LinkageRule.java
package com.ghlzm.iot.alarm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 联动规则实体
 */
@Data
@TableName("linkage_rule")
public class LinkageRule implements Serializable {
      private static final long serialVersionUID = 1L;

      /**
       * 主键ID
       */
      @TableId(type = IdType.AUTO)
      private Long id;

      /**
       * 规则名称
       */
      private String ruleName;

      /**
       * 描述
       */
      private String description;

      /**
       * 触发条件（JSON格式）
       */
      private String triggerCondition;

      /**
       * 动作列表（JSON格式）
       */
      private String actionList;

      /**
       * 状态：0-启用，1-停用
       */
      private Integer status;

      /**
       * 租户ID
       */
      private Long tenantId;

      /**
       * 创建时间
       */
      private LocalDateTime createTime;

      /**
       * 更新时间
       */
      private LocalDateTime updateTime;

      /**
       * 创建人
       */
      private Long createBy;

      /**
       * 更新人
       */
      private Long updateBy;

      /**
       * 逻辑删除：0-未删除，1-已删除
       */
      private Integer deleted;
}

File: spring-boot-iot-ui/src/api/linkageRule.ts
import { request } from './request';
import { buildQueryString } from './query';
import type { ApiEnvelope, IdType } from '../types/api';

/**
 * 联动规则 API
 */

// 联动规则接口定义
export interface LinkageRule {
      id: IdType;
      ruleName: string;
      description: string;
      triggerCondition: string;
      actionList: string;
      status: number;
      tenantId: IdType;
      createTime: string;
      updateTime: string;
      createBy: number;
      updateBy: number;
      deleted: number;
}

export interface LinkageRulePageResult {
      total: number;
      pageNum: number;
      pageSize: number;
      records: LinkageRule[];
}

// 获取规则列表
export const getRuleList = (params?: {
      ruleName?: string;
      status?: number;
}): Promise<ApiEnvelope<LinkageRule[]>> => {
      const queryString = buildQueryString(params);
      const path = queryString ? `/api/linkage-rule/list?${queryString}` : '/api/linkage-rule/list';
      return request<LinkageRule[]>(path, { method: 'GET' });
};

// 分页获取规则列表
export const pageRuleList = (params?: {
      ruleName?: string;
      status?: number;
      pageNum?: number;
      pageSize?: number;
}): Promise<ApiEnvelope<LinkageRulePageResult>> => {
      const queryString = buildQueryString(params);
      const path = queryString ? `/api/linkage-rule/page?${queryString}` : '/api/linkage-rule/page';
      return request<LinkageRulePageResult>(path, { method: 'GET' });
};

// 获取规则详情
export const getRuleById = (id: IdType): Promise<ApiEnvelope<LinkageRule>> => {
      return request<LinkageRule>(`/api/linkage-rule/get/${id}`, { method: 'GET' });
};

// 新增规则
export const addRule = (data: Partial<LinkageRule>): Promise<ApiEnvelope<LinkageRule>> => {
      return request<LinkageRule>('/api/linkage-rule/add', { method: 'POST', body: data });
};

// 更新规则
export const updateRule = (data: Partial<LinkageRule>): Promise<ApiEnvelope<LinkageRule>> => {
      return request<LinkageRule>('/api/linkage-rule/update', { method: 'POST', body: data });
};

// 删除规则
export const deleteRule = (id: IdType): Promise<ApiEnvelope<void>> => {
      return request<void>(`/api/linkage-rule/delete/${id}`, { method: 'POST' });
};
